client/signup/config/steps.js

```diff
@@ -63,5 +63,18 @@ module.exports = {
 		},
 		dependencies: [ 'siteSlug' ],
 		providesDependencies: [ 'theme', 'images' ]
+	},
+
+	'mlb-themes': {
+		stepName: 'mlb-themes',
+		apiRequestFunction: stepActions.setThemeOnSite,
+		dependencies: [ 'siteSlug' ]
+	},
+
+	'mlb-domains': {
+		stepName: 'mlb-domains',
+		apiRequestFunction: stepActions.createSite,
+		providesDependencies: [ 'siteSlug' ],
+		delayApiRequestUntilComplete: true
```

Comment:
Do you mean `delayApiRequestUntilComplete` ? Or the whole definition? (as opposed to using just the `Site` step?

If you meant the latter I can say that at first, I replicated the `domains` step definition, then changed it to be similar to `site` definition. 

The thing is that the way the API will handle the creation of `.mlblogs.com` sites is still to be defined. 
If it's implemented as a free product, then this step may be needing to provide the `domainItem` dependency also instead of just the `siteSlug` and use an API request function like `stepAction.addDomainsItemToCart` instead of `stepAction.createSite`.

And the `mlb-domain` step component would need to have some extra logic for handling submit. (Currently the Site step is handling the submit, saving the step using  `stepAction.createSite` which demands less properties than `stepAction.addDomainItemToCart`  when submitting signup step.